src/google-classroom-extractor/edfi_google_classroom_extractor/api/courses.py

```diff
@@ -20,6 +20,178 @@
+    Returns a list of alias names from a list of aliases with the given scope.
+    Aliases from the API are scoped with a prefix character followed by a colon
+    and then the alias name. For example "d:alias_name_1".
+
+    Parameters
+    ----------
+    aliases: List[str]
+        a list of course alias strings from the API
+    scope: str
+        the scope to filter by
+
+    Returns
+    -------
+    List[str]
+        a list of alias names (scope omitted) filtered by the given scope
+    """
+    result: List[str] = []
+    try:
+        result = [
+            alias.split(":")[1] for alias in aliases if alias.split(":")[0] == scope
+        ]
+    except (ValueError, IndexError):
+        pass
+    return result
+
+
+def _find_first_with_edfilms_prefix(alias_names: List[str]) -> str:
+    """
+    Returns the first alias name in the list prefixed with the
+    Ed-Fi LMS indicator, or an empty string if none are found.
+
+    Parameters
+    ----------
+    alias_names: List[str]
+        a list of alias names
+
+    Returns
+    -------
+    str
+        The first alias name in the list prefixed with the
+        Ed-Fi LMS indicator, or an empty string if none are found.
+    """
+    return next(
+        (alias_name for alias_name in alias_names if alias_name.startswith("EdFiLMS")),
+        "",
+    )
+
+
+def _select_alias(aliases: List[str]) -> str:
+    """
+    Selects the most relevant course alias from a list of course aliases returned by the API
+
+    Parameters
+    ----------
+    aliases: List[str]
+        a list of course alias strings from the API, including the "d:" (domain scope)
+        or "p:" (project scope) prefix
+
+    Returns
+    -------
+    str
+        a (possibly empty) string representing the most relevant alias for the course.
+
+    Notes
+    -----
+    How to find the correct SISSectionIdentifier from CourseAlias results:
+
+    If there are no aliases:
+    SISSectionIdentifier = ""
+
+    Else if there is exactly one alias:
+    SISSectionIdentifier = alias
+
+    Else if there is a domain alias prefixed with "EdFiLMS":
+    SISSectionIdentifier = first domain alias prefixed with "EdFiLMS"
+
+    Else if there is a project alias prefixed with "EdFiLMS":
+    SISSectionIdentifier = first project alias prefixed with "EdFiLMS"
+
+    Else if there are domain aliases:
+    SISSectionIdentifier = first domain alias
+
+    Else if there are project aliases:
+    SISSectionIdentifier = first project alias
+
+    Else:
+    SISSectionIdentifier = ""
+    """
+    if len(aliases) == 0:
+        return ""
+
+    if len(aliases) == 1:
+        try:
+            _, alias_name = aliases[0].split(":")
+            return alias_name
+        except ValueError:
+            return ""
+
+    domain_alias_names: List[str] = _alias_names_for_scope(aliases, "d")
+    edfilms_domain_alias = _find_first_with_edfilms_prefix(domain_alias_names)
+    if edfilms_domain_alias != "":
+        return edfilms_domain_alias
```

Comment:
Unless I'm overlooking something, the output file will have `SectionIdentifier = EdFiLMSxyz` in it - which won't match the SIS. Should we strip off `EdFiLMS`? And if so, should we use any delimiter, e.g. "EdFiLMS.xyz"? No one ever sees the value, so I'm not sure that it is all that useful.

Reply:
Removing prefix.